unmarshaller_test.go

```diff
@@ -156,3 +156,78 @@ func Test_ReadAll(t *testing.T) {
 		t.Fatalf("Expected []sample, but got %T", out)
 	}
 }
+
+func Test_Unmarshaller_Allocation(t *testing.T) {
+    t.Parallel()
+
+	exactHeaders := `field_a,field_b`
+	overlappingHeaders := `field_b,field_c`
+	disjointHeaders := `field_c,field_d`
+
+	var um *Unmarshaller
+	var err error
+	config := &Config{
+		Holder: sample{},
+	}
+
+	// An Unmarshaller should be returned if the file headers match the struct.
+	um, err = config.NewUnmarshaller(csv.NewReader(strings.NewReader(exactHeaders)))
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	if um == nil {
+		t.Fatal("Expected Unmarshaller")
+	}
```

Comment:
```suggestion
	assert.NoError(t, err)
	assert.NotNil(t, um)
```

Reply:
The `assert` package is part of [testify](https://github.com/stretchr/testify), which gocsv/Commando don't use.  This is the normal way of writing stuff using only [Go's `testing`](https://pkg.go.dev/testing) package.  I fully agree that the stdlib testing stuff is verbose and crappy, but I think porting these to testify is out of scope.

Also, our stuff uses a somewhat patched version & I'm wary of causing issues in either repo by depending on different versions.

Reply:
Oh, I see! I didn't realize we were in the commando repo, thanks!